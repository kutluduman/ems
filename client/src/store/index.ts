import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import auth from "./slices/auth";
import employee from "./slices/employee";

export const store = configureStore({
  reducer: {
    auth,
    employee,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;