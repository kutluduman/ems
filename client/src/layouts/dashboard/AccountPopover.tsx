import { alpha } from "@mui/material/styles";
import { useNavigate } from "react-router-dom";
import { useRef, useState } from "react";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import Divider from "@mui/material/Divider";
import IconButton from "@mui/material/IconButton";
import Typography from "@mui/material/Typography";

import { logout, selectEmployee } from "../../store/slices/auth";
import { useAppDispatch, useAppSelector } from "../../store/hooks";
import EmployeeProfileMenu from "../../components/EmployeeProfileMenu";
import UserAvatar from "../../components/UserAvatar";

const AccountPopover = () => {
  const anchorRef = useRef(null);
  const [open, setOpen] = useState(false);
  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const account = useAppSelector(selectEmployee);

  const handleOpen = () => setOpen(true);
  const handleClose = () => setOpen(false);

  return (
    <>
      <IconButton
        ref={anchorRef}
        onClick={handleOpen}
        sx={{
          padding: 0,
          width: 44,
          height: 44,
          ...(open && {
            "&:before": {
              zIndex: 1,
              content: "''",
              width: "100%",
              height: "100%",
              borderRadius: "50%",
              position: "absolute",
              bgcolor: (theme) => alpha(theme.palette.grey[900], 0.72),
            },
          }),
        }}
      >
        <UserAvatar name={account?.name} surname={account?.surname} />
      </IconButton>

      <EmployeeProfileMenu
        open={open}
        onClose={handleClose}
        anchorEl={anchorRef.current}
        sx={{ width: 220 }}
      >
        <Box sx={{ my: 1.5, px: 2.5 }}>
          <Typography variant="subtitle1" noWrap>
            {account?.name} {account?.surname}
          </Typography>
          <Typography variant="body2" sx={{ color: "text.secondary" }} noWrap>
            {account?.email}
          </Typography>
        </Box>

        <Divider sx={{ my: 1 }} />

        <Box sx={{ p: 2, pt: 1.5 }}>
          <Button
            fullWidth
            color="inherit"
            variant="outlined"
            onClick={() => {
              dispatch(logout());
              navigate("/login", { replace: true });
            }}
          >
            Logout
          </Button>
        </Box>
      </EmployeeProfileMenu>
    </>
  );
}
export default AccountPopover;