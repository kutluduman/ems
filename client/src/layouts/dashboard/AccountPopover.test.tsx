import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import { store } from "../../store";
import AccountPopover from "./AccountPopover";

describe("AccountPopover Component", () => {
  test("renders the AccountPopover Component", () => {
    const tree = create(
      <Provider store={store}>
        <Router>
          <AccountPopover />
        </Router>
      </Provider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});