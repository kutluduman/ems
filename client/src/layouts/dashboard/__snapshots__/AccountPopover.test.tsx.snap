// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPopover Component renders the AccountPopover Component 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-k2yn7a-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
  >
    <img
      alt="photoURL"
      className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
      src="https://minimal-kit-react.vercel.app/static/mock-images/avatars/avatar_default.jpg"
    />
  </div>
</button>
`;
